pkg/apis/feeds/v1alpha1/feed_types.go

```diff
@@ -30,7 +30,8 @@ import (
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // Feed connects an event trigger with an action that processes events produced
-// by the trigger.
+// by the trigger. Feeds are building blocks used to implement Flows and are not
```

Comment:
I think a feed is useful as a fundimental block for other platforms. I don't think we should call out there is no expectations to use a feed directly.

Reply:
@n3wscott originally I felt the same, however, there's a direct name association between Flow and Feed - which means that if you use Feed directly, you risk disrupting a flow with the name you used. If we want Feed to be a reusable building block we should change the name association rules.

Reply:
This is true if you edit the feed from a flow, but nothing is stopping you from making the feed object directly.

Flow encapsulates Feed, Channel, and Subscription. But those components are able to be used directly without a Flow.

Reply:
If I make a Feed with name X, i cannot also make a Flow with name X and have it work. To me that means that Feed isn't usable independently. We should change the name association rules so that Flows use GenerateName to create Feeds if we want feeds to be independently usable.

Reply:
I think you are confusing creating a flow and then poking the feed it created for you vs creating a feed. 

I expect a user to be able to create a feed if they want. I also expect them to not touch the feed if it came from a flow.

Reply:
Oh I see. Yes that is a valid issue, and we can solve that with robotic naming. But my original concern with blocking users from creating feeds still stands. flows are an impl of the usage of a feed I think

Reply:
I think the name collision is a _bug_, not an intentional behavior. We should probably be using `generateName` when creating the Feed, Channel, and Subscription names from the Flow (or else document that they share naming, ala Service -> Route and Configuration in `knative/serving`).

In this case, I don't think there's substantial harm from using `generateName`, since we intend for the Feed and Channel resources to be managed completely by the Flow. I could be convinced that we should instead document the collision problems, but I don't think nice names for Feeds is a particularly high priority (unlike names for Routes, which are something that a user might type into a web browser or some other application's config).